constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java

```diff
@@ -24,45 +24,58 @@
 import static org.jboss.pnc.enums.BuildProgress.PENDING;
 
 /**
- * Status of a running or finished build.
+ * Status of a running/completed Build/GroupBuild.
+ * 
+ * The status class is shared by both Build and GroupBuild dtos, which are published through UI/WebSocket.
  *
  * @author Jakub Bartecek &lt;jbartece@redhat.com&gt;
  *
  */
 public enum BuildStatus {
     /**
-     * Build completed successfully.
+     * Build/GroupBuild has completed successfully. The dependant Builds can be scheduled.
      */
     SUCCESS(FINISHED, true),
 
     /**
-     * Build failed.
+     * Build/GroupBuild has failed. The status is propagated through dependants as REJECTED_FAILED_DEPENDENCIES.
+     *
+     * A GroupBuild is reported FAILED with is at least one underlying Build with the final status of (FAILED, REJECTED,
+     * SYSTEM_ERROR)
      */
     FAILED(FINISHED, false),
 
     /**
-     * A build has been requested (possibly via dependencies) but no actual build happened as it was not required (no
-     * updates).
+     * A Build has been requested (possibly via dependencies) but no actual build happened as it was not required (no
+     * updates to underlying BuildConfig and Configs of dependencies (depends in IMPLICIT/EXPLICIT dependency check)).
```

Comment:
I would make it more clear by "(rebuild is required when there is ..."

I think rebuild is required when:
 * the BC changed
 * an explicit/implicit (transitive) dependency BC was rebuilt
 * an explicit/implicit (transitive) dependency BC changed
(if this is not true, do we have good implementation?)

Reply:
Sure I can make it much more clear.

Reply:
Regarding rebuild possibilities, there are three modes:

- **Explicit**
    - The build config has been modified since the last successful build
    - A build config, explicitly defined as a dependency of this one, has been modified
    - since the last successful build
    - There exists a newer, successful build of a build config explicitly defined as a dependency of this one
- **Implicit**
    - Explicit criteria plus:
    - There exists a newer version of an implicit dependency (automatically
    - captured from sources such as Indy, MRRC or Maven Central)
- **Force**
    - Always

Reply:
@matedo1 that's the definition we use in the UI right? Hopefully someone with a better idea of the backend implementation, e.g. Honza or Matej can verify that our description is still accurate.

Reply:
@alexcreasy yes, the definition is coming from the UI

Reply:
@alexcreasy @matedo1 WDYT about changing the Implicit description a little. For me, it's a little confusing with what we have now. 
I'd add a clause similar to this: 
     - implicit dependencies are also Builds that produced artifacts used during build process

Reply:
@michalovjan Martin's paste is a bit innacurate, it is missing the first line:

"Rebuild a previously successful build config when:"

So the bullet points are the minimum criteria which must be met for a rebuild to take place, for each rebuild mode. Your description of "implicit builds" makes no sense in that context, as it's not a criteria that must be met for a rebuild to take place.

If you want to rework the text some more so that your clarification fits in overall, by all means feel free. I don't have a complete understanding of rebuild modes, as they're exclusively backend features, so I'm not the best person to write / critique. Small modifications need to fit the overal context though.

Here's the full text from the UI popup menu, for reference:

```
Rebuild a previously successful build config when:

**Explicit**
- The build config has been modified since the last successful build
- A build config, explicitly defined as a dependency of this one, has been modified since the last successful build
- There exists a newer, successful build of a build config explicitly defined as a dependency of this one.

**Implicit**
- Explicit criteria plus:
- There exists a newer version of an implicit dependency (automatically captured from sources such as Indy, MRRC or Maven Central)

**Force**
- Always
```

Reply:
Oh on second reading, I'm wondering if you were just proposing to change the implicit clause rather than add a new one?

Reply:
Yes, only implicit clause. I thought maybe instead of using parentheses, use further indentation.

Rebuild a previously successful build config when:

**Explicit**
- The build config has been modified since the last successful build
- A build config, explicitly defined as a dependency of this one, has been modified since the last successful build
- There exists a newer, successful build of a build config explicitly defined as a dependency of this one.

**Implicit**
- Explicit criteria plus:
- There exists a newer version of an implicit dependency that can be:
        - automatically captured from sources such as Indy, MRRC or Maven Central
        - a Build which produced artifacts that were used for alignment during build process

**Force**
- Always

Reply:
^ @alexcreasy @matedo1